pkg/common/utils_test.go

```diff
@@ -1220,6 +1228,14 @@ func TestCodeForError(t *testing.T) {
 			inputErr: &googleapi.Error{Code: http.StatusBadRequest, Message: "User error with bad request"},
 			expCode:  codes.InvalidArgument,
 		},
+		{
+			name: "googleapi.Error that wraps apierror.APIError, that wraps googleapi.Error",
```

Comment:
I thought the errors we receive are "github.com/googleapis/gax-go/v2/apierror" APIError, which wrap a googleAPI error? When will "googleapi.Error that wraps apierror.APIError, that wraps googleapi.Error" happen?

Can you add other test cases, for when api.APIError wraps other types of errors that aren't googleapi errors?

Reply:
You are right! We are not necessarily wrapping a third error here. The way the error gets created is by passing in googleAPI error to a function. I think we benefit for adding more test cases to deal with the different ways the APIError can be structured.

Reply:
Added more test cases.